### PR TITLE
Throw error if `debug.yaml` not found

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
Shows error this error if `swift test` is run before `swift build`
 ```
error: build the package using `swift build` before running tests
``` 

instead of 

```
<unknown>:0: error: unable to attach DB: unable to open database
error: exit(1): swift-build-tool -f /Users/aciid/mycode/test/MyPackage/.build/debug.yaml test
```

### CODE REVIEW

**Comment by czechboy0** [comment]
Is there a reason to not automatically run `swift build` as the first step of `swift test`? To me running build, then test, then changing some test code and running test again - I'd assume that the changes would be applied. But since testing doesn't rebuild it doesn't, which initially confused me and I wonder whether it's not going to be confusing to newcomers as well. 

I'd probably propose to automatically run build as the first step of test. Or is there a reason not to?

**Comment by aciidb0mb3r** [comment]
agreed, `Utilities/bootstrap test` also builds if not already built instead of exiting with error.
@mxcl suggestions?

**Comment by mxcl** [comment]
@czechboy0 we may change it. As it stands things are more flexible if you can control when the build happens entirely.

File: Sources/swift-test/main.swift [hunk 1]
```diff
@@ -24,6 +24,8 @@ do {
     setenv("SPM_INSTALL_PATH", dir.build, 0)
 
     let yamlPath = Path.join(dir.build, "debug.yaml")
+    if !yamlPath.exists { throw Error.DebugYAMLNotFound }
```
Comment (mxcl) [inline Sources/swift-test/main.swift]: `guard` seems more appropriate to me.

Comment (aciidb0mb3r) [inline Sources/swift-test/main.swift]: Fixed

File: Sources/swift-test/Error.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+extension Error: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .DebugYAMLNotFound:
+            return "build the package using `swift build` before running tests"
+        }
+    }
+}
```
Comment (keith) [inline Sources/swift-test/Error.swift]: Looks like there's no newline at the end of this file.

Comment (aciidb0mb3r) [inline Sources/swift-test/Error.swift]: fixed

**Comment by czechboy0** [comment]
@mxcl Makes sense. The way I think of it is in Xcode, the "easier" action is Run and Test, which actually mean Build+Run and Build+Test. Then you can trigger Run Without Building & Test Without Building manually, but that's more work.
In the same way I'd imagine we'd change the default here to build before testing, but we'd also add a flag to just run, something like `swift build --test --no-rebuild` or similar.

**Pull request merged**